protos/sources/vega/data_source.proto

```diff
@@ -54,6 +56,39 @@ message DataSourceSpecConfiguration {
   repeated vega.data.v1.Filter filters = 2;
 }
 
+// Specifies a data source that derives its content from calling a read method on an Ethereum contract.
+message DataSourceSpecEthCall {
+  // Ethereum address of the contract to call.
+  string address = 1;
+  // The ABI of that contract.
+  google.protobuf.ListValue abi = 2;
+  // Name of the method on the contract to call.
+  string method = 3;
+  // List of arguments to pass to method call.
+  // Protobuf 'Value' wraps an arbitrary JSON type that is mapped to an Ethereum type according to the ABI.
+  repeated google.protobuf.Value args = 4;
+  // Conditions for determining when to call the contract method.
+  EthCallTrigger trigger = 5;
+}
+
+// Determines when the contract method should be called.
+message EthCallTrigger {
```

Comment:
Can we call this simply a `Trigger`?

Reply:
Well we could I guess but it is kind of tied explicitly to Ethereum, e.g. `Trigger when the Ethereum time is greater or equal to this time`

Reply:
All right